rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -21,6 +21,51 @@
 
 namespace
 {
+
+// Foxy compatibility - copied from Galactic version of rmw/time.c
+typedef rcutils_duration_value_t rmw_duration_t;
+// we can't use the Galactic version of this constant because Foxy RMWs won't understand it
+static const rmw_time_t RMW_DURATION_INFINITE {0LL, 0LL};
+
+
+rmw_duration_t
+rmw_time_total_nsec(const rmw_time_t time)
+{
+  static const uint64_t max_sec = INT64_MAX / RCUTILS_S_TO_NS(1);
+  if (time.sec > max_sec) {
+    // Seconds not representable in nanoseconds
+    return INT64_MAX;
+  }
+
+  const int64_t sec_as_nsec = RCUTILS_S_TO_NS(time.sec);
+  if (time.nsec > (uint64_t)(INT64_MAX - sec_as_nsec)) {
+    // overflow
+    return INT64_MAX;
+  }
+  return sec_as_nsec + time.nsec;
+}
+
+bool
+rmw_time_equal(const rmw_time_t left, const rmw_time_t right)
+{
+  return rmw_time_total_nsec(left) == rmw_time_total_nsec(right);
+}
+
+rmw_time_t
+rmw_time_from_nsec(const rmw_duration_t nanoseconds)
+{
+  if (nanoseconds < 0) {
+    return (rmw_time_t)RMW_DURATION_INFINITE;
```

Comment:
Please do not use C-Style casts. This should be `static_cast<rmw_time_t>(RMW_DURATION_INFINITE)`